var/spack/repos/builtin/packages/thrift/package.py

```diff
@@ -34,78 +35,123 @@ class Thrift(Package):
+        variant("qt5", default=False, description="Build with Qt5 support")
+        variant("zlib", default=False, description="Build with ZLIB support")
+    variant("c_glib", default=False, description="Build C (GLib) library")
     variant("java", default=False, description="Build support for java")
+    variant("javascript", default=False, description="Build Javascript library")
+    variant("nodejs", default=False, description="Build NodeJS library")
     variant("python", default=True, description="Build support for python")
 
-    depends_on("c", type="build")  # generated
-    depends_on("cxx", type="build")  # generated
-
-    depends_on("pkgconfig", type="build")
-    depends_on("autoconf", type="build")
-    depends_on("automake", type="build")
-    depends_on("libtool", type="build")
-    depends_on("boost@1.53:")
-
-    # TODO: replace this with an explicit list of components of Boost,
-    # for instance depends_on('boost +filesystem')
-    # See https://github.com/spack/spack/pull/22303 for reference
-    depends_on(Boost.with_default_variants)
-    depends_on("bison", type="build")
-    depends_on("flex", type="build")
-    depends_on("openssl")
-
-    # Variant dependencies
-    depends_on("zlib-api", when="+c")
-    depends_on("libevent", when="+c")
-
-    depends_on("java@7:", when="+java")
-    depends_on("ant", when="+java")
-
-    extends("python", when="+python")
-    depends_on("python@:3.11.9", when="+python")
-    depends_on("py-setuptools", type=("build", "run"), when="+python")
-    depends_on("py-six@1.7.2:", type=("build", "run"), when="@0.10.0:+python")
-    depends_on("py-tornado", type=("build", "run"), when="+python")
-    depends_on("py-twisted", type=("build", "run"), when="+python")
-    depends_on("py-zope-interface", type=("build", "run"), when="+python")
-    depends_on("py-pure-sasl", type=("build", "run"), when="+python")
-    depends_on("scons", type=("build", "run"), when="+python")
-    depends_on("gmake", type="build")
+    build_system("cmake", "autotools", default="autotools")
+
+    with default_args(type="build"):
+        depends_on("c")
+        depends_on("cxx")
+
+        depends_on("pkgconfig")
+        depends_on("bison")
+        depends_on("flex")
+
+        with when("build_system=cmake"):
+            depends_on("cmake@3.4:", when="@0.13:")
+            depends_on("cmake@3.1:", when="@0.11:")
+            depends_on("cmake@2.8.12:")
+
+        with when("build_system=autotools"):
+            depends_on("autoconf")
+            depends_on("automake")
+            depends_on("libtool")
+
+    depends_on("glib@2:", when="+c_glib")
+    depends_on("openssl", when="+openssl")
+    depends_on("libevent@2:", when="+libevent")
+    depends_on("qt5", when="+qt5")
+    depends_on("zlib-api@1.2.3:", when="+zlib")
+
+    with when("+java"):
+        depends_on("ant@1.8:", type="build")
+        depends_on("gradle", type="build")
+        depends_on("java@7:")
```

Comment:
It seems like CMake only cares about Gradle. Autotools checks for both and fails if Ant is missing